### PR TITLE
ENH: faster whitening matrix calculation for sm.GLS()

### LINKED ISSUE #8372: ENH: faster whitening matrix calculation for sm.GLS()
`sm.GLS()` starts by calculating a whitening matrix, which the code calls `cholsigmainv`:

`cholsigmainv = np.linalg.cholesky(np.linalg.inv(sigma)).T`

But if you switch the order of operations so the Cholesky decomposition goes first (which is mathematically valid, if you get rid of the transpose), you only have to invert a lower triangular matrix instead of a dense matrix, which can be done efficiently with Lapack's [`dtrtri()`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.linalg.lapack.dtrtri.html). 

`scipy.linalg.cholesky(..., lower=True)` also gives a slight speed-up over `np.linalg.cholesky(...)`, as does adding `overwrite_c=True` to `dtrtri()`. 

So I'd propose switching [this line](https://github.com/statsmodels/statsmodels/blob/edd843f0afb37b30cb503709e2d4ff4bc57f9c04/statsmodels/regression/linear_model.py#L181) to:

```
from scipy.linalg import cholesky
from scipy.linalg.lapack import dtrtri
cholsigmainv = dtrtri(cholesky(sigma, lower=True), lower=True, overwrite_c=True)[0]
```

This formulation is about four and a half times faster for a random 10,000 x 10,000 covariance matrix:

```
import numpy as np
from scipy.linalg import cholesky
from scipy.linalg.lapack import dtrtri
from timeit import timeit

arr = np.random.RandomState(0).random(size=(10000, 1000))
sigma = np.cov(arr) + 1e-8 * np.eye(len(arr))
timeit(lambda: np.linalg.cholesky(np.linalg.inv(sigma)).T, number=1)  # 304.3108371179551 seconds
timeit(lambda: dtrtri(cholesky(sigma, lower=True), lower=True, overwrite_c=True), number=1)  # 68.95121738314629 seconds
```

Calculating the whitening matrix becomes a major bottleneck for GLS when there are a lot of data points, and this enhancement should help overcome that.

### PR DESCRIPTION
**Pull request by Wainberg**
- [x] closes #8372
- [ ] tests added / passed. 
- [x] code/documentation is well formatted.  
- [x] properly formatted commit message. See [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message).

### CODE REVIEW

File: statsmodels/regression/linear_model.py [hunk 1]
```diff
@@ -178,7 +179,12 @@ def _get_sigma(sigma, nobs):
         if sigma.shape != (nobs, nobs):
             raise ValueError("Sigma must be a scalar, 1d of length %s or a 2d "
                              "array of shape %s x %s" % (nobs, nobs, nobs))
-        cholsigmainv = np.linalg.cholesky(np.linalg.inv(sigma)).T
+        cholsigmainv, info = dtrtri(cholesky(sigma, lower=True), 
+                                    lower=True, overwrite_c=True)
+        if info > 0:
+            raise np.linalg.LinAlgError('Singular matrix')
```
Comment (bashtage) [inline statsmodels/regression/linear_model.py:185]: Can you chenge the exception to explaing that sigma is singular.

**Comment by bashtage** [comment]
Got a PEP 8 violation.

**Comment by Wainberg** [comment]
I fixed the PEP 8 violation and changed the error message to "Cholesky decomposition of sigma yields a singular matrix".

**Comment by bashtage** [comment]
Great, thanks.

**Pull request merged**